validator/remote/src/main/java/tech/pegasys/teku/validator/remote/FailoverValidatorApiHandler.java

```diff
@@ -399,15 +398,15 @@ private <T> SafeFuture<T> makeRequestToFailoversUntilSuccess(
       final ValidatorApiChannelRequest<T> request,
       final String method,
       final Map<RemoteValidatorApiChannel, Throwable> capturedExceptions) {
-    final Iterator<RemoteValidatorApiChannel> failoverDelegates =
+    final Iterator<? extends RemoteValidatorApiChannel> failoverDelegates =
         beaconNodeReadinessManager.getFailoversInOrderOfReadiness();
     return makeRequestToFailoversUntilSuccess(
         failoverDelegates.next(), failoverDelegates, request, method, capturedExceptions);
   }
 
   private <T> SafeFuture<T> makeRequestToFailoversUntilSuccess(
       final RemoteValidatorApiChannel currentFailoverDelegate,
-      final Iterator<RemoteValidatorApiChannel> failoverDelegates,
+      final Iterator<? extends RemoteValidatorApiChannel> failoverDelegates,
```

Comment:
Is there a reason for having few changes of those `? extends`

Reply:
Yes, the root of it is
```java
final List<? extends RemoteValidatorApiChannel> failoverValidatorApis =
    failoverEndpoints.stream()
        .map(
            endpoint ->
                RemoteValidatorApiHandler.create(
                    endpoint,
                    okHttpClient,
                    spec,
                    validatorConfig.isValidatorClientUseSszBlocksEnabled(),
                    asyncRunner))
        .toList();
```

Reply:
This is not the only place it happens, when we want to use the higher level interface as type

Reply:
Got it. I don't mind it. Makes sense.